diff -uNr libidn2-2.3.8/Makefile.in libidn2-2.3.8-notools/Makefile.in
--- libidn2-2.3.8/Makefile.in	2025-03-08 16:51:42.000000000 -0400
+++ libidn2-2.3.8-notools/Makefile.in	2025-07-11 21:05:01.926750962 -0300
@@ -1772,7 +1772,7 @@
 unistringtests_LIBOBJS = @unistringtests_LIBOBJS@
 unistringtests_LTLIBOBJS = @unistringtests_LTLIBOBJS@
 unistringtests_WITNESS = @unistringtests_WITNESS@
-SUBDIRS = gl unistring lib src examples fuzz po $(am__append_1) tests
+SUBDIRS = gl unistring lib examples fuzz po $(am__append_1) tests
 
 # version-stamp
 EXTRA_DIST = cfg.mk maint.mk CONTRIBUTING.md DEPENDENCIES.md README.md \
